### ISSUE TITLE
Incorrect Balancing Mode for Internal Load Balancer

### ISSUE THREAD

**Issue by siddharth-mistry** [terraform-google-modules/terraform-google-lb-internal#146]
### TL;DR

Need to set `balancing_mode` under dynamic block for backends in `google_compute_region_backend_service` in main.tf to `CONNECTION`

### Expected behavior

The Terraform module should either:

1. Automatically set the `balancing_mode` to `CONNECTION` for internal load balancers, preventing the error.
2. Option to set `balancing_mode` to `CONNECTION` via variable

### Observed behavior

When using the Terraform module to create an internal load balancer (ILB) on GCP, an error occurs as default value for `balancing_mode` is set to `UTILIZATION` by default.

Error message:
```
Error: Error creating RegionBackendService: googleapi: Error 400: Invalid value for field 'resource.backends[0].balancingMode': 'UTILIZATION'. Balancing mode must be CONNECTION for an INTERNAL backend service., invalid
```

### Terraform Configuration

```hcl
module "gce_ilb" {
  source                       = "GoogleCloudPlatform/lb-internal/google"
  version                      = ">= 4.4.0"
  project                      = var.project_id
  region                       = var.region
  name                         = "tf-ilb"
  network                      = "testnetwork"
  subnetwork                   = "testsubnet"
  ports                        = ["xxxx"]
  health_check                 = local.health_check
  source_tags                  = ["xxxxxx"]
  target_tags                  = ["xxxxxx"]
  backends = [
    for x in google_compute_instance_group.test_ig[*] : {
      group = x.id
      description = ""
      failover = false
    }
  ]
  create_backend_firewall      = "true"
  create_health_check_firewall = "true"
}
```

### Terraform Version

```sh
Terraform v1.8.5
```

### Additional information

Need below change in main.tf
```
  dynamic "backend" {
    for_each = var.backends
    content {
      group          = lookup(backend.value, "group", null)
      description    = lookup(backend.value, "description", null)
      failover       = lookup(backend.value, "failover", null)
      balancing_mode = lookup(backend.value,"balancing_mode,"CONNECTION")
    }
  }
  ```

**Comment by imrannayer** [comment]
Fixed in #148
